File: Algorithms/Swift/Easy/83_Remove_Duplicates_from_Sorted_List.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.next = nil
 *     }
 * }
 */
class Solution {
    func deleteDuplicates(head: ListNode?) -> ListNode? {
        if head == nil || head!.next == nil {
            return head
        }

        var current = head
        while current!.next != nil {
            let value = current!.val
            if current!.next!.val == value {
                var node = current!.next
                while node != nil && node!.val == value {
                    node = node!.next
                }
                current!.next = node
            } else {
                current = current!.next
            }
        }

        return head
    }
}

File: Algorithms/JavaScript/Easy/83_Remove_Duplicates_from_Sorted_List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    if (head === null) {
        return null
    }

    var currentNode = head
    while (currentNode.next !== null) {
        var value = currentNode.val
        if (currentNode.next.val == value) {
            var node = currentNode.next
            while (node !== null && node.val === value) {
                node = node.next
            }
            currentNode.next = node
        } else {
            currentNode = currentNode.next
        }
    }

    return head
};
